x-pack/plugins/task_manager/server/task_claimers/strategy_update_by_query.ts

```diff
@@ -57,52 +56,47 @@ interface OwnershipClaimingOpts {
   taskMaxAttempts: Record<string, number>;
 }
 
-export function claimAvailableTasksUpdateByQuery(
+export async function claimAvailableTasksUpdateByQuery(
   opts: TaskClaimerOpts
-): Observable<ClaimOwnershipResult> {
+): Promise<ClaimOwnershipResult> {
   const { getCapacity, claimOwnershipUntil, batches, events$, taskStore } = opts;
   const { definitions, unusedTypes, excludedTaskTypes, taskMaxAttempts } = opts;
   const initialCapacity = getCapacity();
-  return from(batches).pipe(
-    mergeScan(
-      (accumulatedResult, batch) => {
-        const stopTaskTimer = startTaskTimer();
-        const capacity = Math.min(
-          initialCapacity - accumulatedResult.stats.tasksClaimed,
-          isLimited(batch) ? getCapacity(batch.tasksTypes) : getCapacity()
-        );
-        // if we have no more capacity, short circuit here
-        if (capacity <= 0) {
-          return of(accumulatedResult);
-        }
-        return from(
-          executeClaimAvailableTasks({
-            claimOwnershipUntil,
-            size: capacity,
-            events$,
-            taskTypes: isLimited(batch) ? new Set([batch.tasksTypes]) : batch.tasksTypes,
-            taskStore,
-            definitions,
-            unusedTypes,
-            excludedTaskTypes,
-            taskMaxAttempts,
-          }).then((result) => {
-            const { stats, docs } = accumulateClaimOwnershipResults(accumulatedResult, result);
-            stats.tasksConflicted = correctVersionConflictsForContinuation(
-              stats.tasksClaimed,
-              stats.tasksConflicted,
-              initialCapacity
-            );
-            return { stats, docs, timing: stopTaskTimer() };
-          })
-        );
-      },
-      // initialise the accumulation with no results
-      accumulateClaimOwnershipResults(),
-      // only run one batch at a time
-      1
-    )
-  );
+
+  let accumulatedResult = getEmptyClaimOwnershipResult();
+  const stopTaskTimer = startTaskTimer();
+  for (const batch of batches) {
```

Comment:
Previously when using observables, we run the claim query for each batch and emit the claim result as an observable as they become available. With this switch to promises, we are iterating over the batches and accumulating the claim result before returning the full claim result as a promise.